kytos/core/user.py

```diff
@@ -25,35 +27,54 @@ def dict(self, **kwargs) -> dict:
         return values
 
 
+class HashSubDoc(BaseModel):
+    """HashSubDoc. Parameters for hash.scrypt function"""
+    salt: bytes = None
+    n: int = 8192
+    r: int = 8
+    p: int = 1
+
+
 class UserDoc(DocumentBaseModel):
     """UserDocumentModel."""
 
     username: constr(min_length=1, max_length=64, regex="^[a-zA-Z0-9_-]+$")
+    hash: HashSubDoc = HashSubDoc(**{})
     state: Literal['active', 'inactive'] = 'active'
-    password: constr(min_length=8)
     email: EmailStr
+    password: constr(min_length=8, max_length=64)
 
-    @validator('password')
+    @root_validator
     # pylint: disable=no-self-argument
-    def validate_password(cls, password):
+    def validate_password(cls, values):
         """Check if password has at least a letter and a number"""
         upper = False
         lower = False
         number = False
-        for char in password:
+        for char in values['password']:
```

Comment:
There are many things to tackle here, so I am going to list them and explained the applied solution:
- `controller.auth._create_superuser()` was giving a false positive. Changed the return string so it shows that an error has ocurred and its description.
Eg. `Out[1]: 'Error: email: value is not a valid email address`
- `@root_validator` failed validations. Decided to discard it because of the list nature of `ValidationError` reports (password validation error message was on its own). Changed to using `@validator('salt', pre=True, always=True)` and `HashSubDoc` is being initiated from `UserController.create_user()`.
- Error were not helpful, at all. Introducing new static method `Auth.error_msg` to get a request friendly error message, otherwise is presented as raw: 
->raw: `"description": "3 validation errors for UserDoc\nusername\n  string does not match regex \"^[a-zA-Z0-9_-]+$\" (type=value_error.str.regex; pattern=^[a-zA-Z0-9_-]+$)\nemail\n  value is not a valid email address (type=value_error.email)\npassword\n  ('Password should contain: ', ' Minimun 8 characters,', ' at least one upper case character,', ' at least 1 numeric character [0-9].') (type=value_error)"`
->current: `"description": "username: ensure this value has at least 1 characters; email: value is not a valid email address; password: value should contain minimun 8 characters, at least one upper case character, at least 1 numeric character [0-9]",`

Reply:
@Alopalao, fantastic. Great enhancements.

Reply:
I'll go ahead and mark this thread as resolved.